qiskit_experiments/base_analysis.py

```diff
@@ -87,6 +87,7 @@ def run(
         analysis_options = self._default_options()
         analysis_options.update_options(**options)
         analysis_options = analysis_options.__dict__
+        experiment_data._metadata["analysis_options"] = analysis_options
```

Comment:
This can be moved to the base experiment. I prefer managing metadata in a single place, rather than distributing the functionality.

Reply:
@yaelbh do you know if the terra AnalysisResult base class have a general metadata field we could store these analysis options in instead of in the experiment metadata? If so maybe we could move the 'runtime' analysis options used there (which this was intended to be able to capture), and then as @nkanazawa1989 suggests, keeping hte experiment metadata contained to base experiment class.

Reply:
I see that you already moved everything to the same location, as @nkanazawa1989 suggested. In PR 5499, in the file named `analysis_result.py`, the word "metadata" does not appear.